src/app/view/components/_common/dropdown-component/dot-dropdown.component.spec.ts

```diff
@@ -0,0 +1,153 @@
+import { ComponentFixture, tick, fakeAsync } from '@angular/core/testing';
+import { Component, Input, DebugElement } from '@angular/core';
+import { DOTTestBed } from '../../../../test/dot-test-bed';
+import { DotDropdownComponent } from './dot-dropdown.component';
+import { By } from '@angular/platform-browser';
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
+
+@Component({
+    selector: 'dot-test-host-component',
+    template: `<dot-dropdown-component
+            [gravatar]="gravatar"
+            [icon]="icon"
+            [title]="title"
+            [disabled]="disabled"></dot-dropdown-component>`
+})
+class DotTestHostComponent {
+    disabled: boolean;
+    gravatar: string;
+    icon: string;
+    title: string;
+
+    constructor() {
+        this.icon = 'icon';
+        this.gravatar = 'test@test.com';
+        this.title = 'test';
+    }
+}
+
+@Component({
+    selector: 'dot-gravatar',
+    template: ''
+})
+class MockGravatarComponent {
+    @Input()
+    email;
+}
+
+@Component({
+    selector: 'dot-icon-button',
+    template: ''
+})
+class MockDotIconButtonComponent {
+    @Input()
+    icon: string;
+    @Input()
+    disabled?: boolean;
+}
+
+function executeEnabled(
+    elem: DebugElement,
+    hostFixture: ComponentFixture<DotTestHostComponent>,
+    de: DebugElement,
+    comp: DotDropdownComponent
+) {
+    elem.nativeElement.click();
+    hostFixture.detectChanges();
+    const content = de.query(By.css('.dropdown-content'));
+    expect(content).toBeTruthy();
+    expect(elem).toBeTruthy();
+    expect(comp.toggle.emit).toHaveBeenCalledWith(true);
+}
+
+function executeDisabled(elem: DebugElement, de: DebugElement) {
+    elem.nativeElement.click();
+    tick(1);
+    const content = de.query(By.css('.dropdown-content'));
+    expect(content).toBeFalsy();
+    expect(elem).toBeTruthy();
+}
+
+describe('DotDropdownComponent', () => {
+    let hostFixture: ComponentFixture<DotTestHostComponent>;
+    let hostDe: DebugElement;
+    let hostComp: DotTestHostComponent;
+
+    let comp: DotDropdownComponent;
+    let de: DebugElement;
+
+    beforeEach(() => {
```

Comment:
We need async here, because we're using external html for the template more info: https://angular.io/guide/testing#calling-compilecomponents